### PR TITLE
Update smallrye-graphql to add support for GraphQL unions

### PR DESCRIPTION
**Pull request by craig-day**
Updating SmallRye GraphQL to get support for GraphQL unions.

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
         <smallrye-health.version>3.2.1</smallrye-health.version>
         <smallrye-metrics.version>3.0.5</smallrye-metrics.version>
         <smallrye-open-api.version>2.1.22</smallrye-open-api.version>
-        <smallrye-graphql.version>1.6.0</smallrye-graphql.version>
+        <smallrye-graphql.version>1.6.2-SNAPSHOT</smallrye-graphql.version>
```
Comment (craig-day) [inline bom/application/pom.xml:47]: This is my testing locally and will be changed before this PR is marked as ready for review.

**Review by phillip-kruger — APPROVED** [review]
LGTM. Let's add the guide too.

**Comment by phillip-kruger** [comment]
@craig-day Thanks a lot for this. The Guide source is here: https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/smallrye-graphql.adoc

I'll go ahead and release 1.6.2 in the mean time.

**Comment by craig-day** [comment]
Thanks @phillip-kruger, I added a rather lengthy section to the guide for interfaces and unions. Let me know if it's too verbose or if you want the examples to be different.

**Review by phillip-kruger — APPROVED** [review]
This looks great!! Thanks @craig-day . Can you squash your commits and then we should be good to go.

**Comment by craig-day** [comment]
@phillip-kruger done! squashed and rebased on latest main

**Comment by phillip-kruger** [comment]
Thanks @craig-day ! There are some related test failures. I'll also have a look at them a bit later

**Comment by craig-day** [comment]
@phillip-kruger Looks like some more class loading problems because I added a reference to `TestUnion` in the `TestResource` class, so I had to add the new classes anywhere `TestResource` was used.

**Comment by phillip-kruger** [comment]
Awesome ! Thanks @craig-day ! Let's see what CI is doing

**Review by gsmet — CHANGES_REQUESTED** [review]
@phillip-kruger can you release a Jakarta version and update the version here:
https://github.com/quarkusio/quarkus/blob/main/jakarta/rewrite.yml#L429
in this PR?

Otherwise the Jakarta build will fail.

Thanks!

**Comment by phillip-kruger** [comment]
@jmartisk is already on it :)

**Comment by jmartisk** [comment]
The Jakarta upgrade is in now.

**Review by gsmet — APPROVED** [review]

**Comment by phillip-kruger** [comment]
@gsmet The test failures seem unrelated ?

**Comment by gsmet** [comment]
Better wait for the report :).

**Pull request merged**